lib/elixir/lib/range.ex

```diff
@@ -40,9 +40,13 @@ defmodule Range do
   not materialize the whole list of integers.
   """
 
-  defstruct first: nil, last: nil
+  @enforce_keys [:first, :last, :step]
+  defstruct first: nil, last: nil, step: nil
 
-  @type t :: %__MODULE__{first: integer, last: integer}
+  @type first :: integer
```

Comment:
I wonder if this opens the opportunity to consider ranges that are floats? Given that the step is now explicit it would seem to remove the ambiguity associated with enumerating float ranges?